pmd-core/src/main/java/net/sourceforge/pmd/lang/antlr/AntlrBaseParser.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.antlr;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.antlr.v4.runtime.Lexer;
+
+import net.sourceforge.pmd.lang.Parser;
+import net.sourceforge.pmd.lang.ParserOptions;
+import net.sourceforge.pmd.lang.TokenManager;
+import net.sourceforge.pmd.lang.ast.AntlrBaseNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.ast.ParseException;
+
+/**
+ * Generic Antlr parser adapter for all Antlr parsers.
+ */
+public abstract class AntlrBaseParser implements Parser {
+
+    protected final ParserOptions parserOptions;
+
+    public AntlrBaseParser(final ParserOptions parserOptions) {
+        this.parserOptions = parserOptions;
+    }
+
+    @Override
+    public ParserOptions getParserOptions() {
+        return parserOptions;
+    }
+
+    @Override
+    public TokenManager getTokenManager(final String fileName, final Reader source) {
+        try {
+            return new AntlrTokenManager(getLexer(source), fileName);
+        } catch (final IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public Node parse(final String fileName, final Reader source) throws ParseException {
+        AntlrBaseNode rootNode = null;
+        try {
+            rootNode = getRootNode(getParser(getLexer(source)));
+        } catch (final IOException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            e.printStackTrace();
```

Comment:
don't eat this up

Reply:
I'll change this to throw a ParseException when merging.